hi-ml-cpath/src/health_cpath/utils/plots_utils.py

```diff
@@ -73,6 +73,26 @@ def save_pr_curve(results: ResultsType, figures_dir: Path, stage: str = '') -> N
         logging.warning("The PR curve plot implementation works only for binary cases, this plot will be skipped.")
 
 
+def save_roc_curve(results: ResultsType, figures_dir: Path, stage: str = '') -> None:
+    """Plots and saves ROC curve figure in its dedicated directory. This implementation
+    only works for binary classification.
+''
+    :param results: Dict of lists that contains slide_level results
+    :param figures_dir: The path to the directory where to save the figure
+    :param stage: Test or validation, used to name the figure. Empty string by default.
+    """
+    true_labels = [i.item() if isinstance(i, Tensor) else i for i in results[ResultsKey.TRUE_LABEL]]
```

Comment:
This fragment is repeated very often (line 86, 104, 105). Add a function for that?

Reply:
Thanks for the suggestion, I have now replaced these lines with function calls